pkg/secrets/secrets.go

```diff
@@ -47,7 +47,7 @@ var secretsFile = "secrets.json"
 
 // secretNameRegexp matches valid secret names
 // Allowed: 64 [a-zA-Z0-9-_.] characters, and the start and end character must be [a-zA-Z0-9]
-var secretNameRegexp = regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9_.-]*[a-zA-Z0-9]$`)
+var secretNameRegexp = regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9_.-]*[a-zA-Z0-9]?$`)
```

Comment:
Doesn’t this now accept `a-` ? At least per the comment, that seems unintended.

Reply:
oh no i think it does :( regex is tricky

Reply:
FWIW https://github.com/containers/image/blob/7c8c9a604200ea0593d55a75e448b05a82f90ce6/docker/reference/regexp.go#L18-L20 is one way to express this.